docs/adr-2024-04-add-support-to-esm.md

```diff
@@ -0,0 +1,41 @@
+# ADR: Add support to ECMAScript Modules (ESM)
+
+## Context
+
+Our build and bundling setup currently utilizes CommonJS modules along with webpack and the dynamic-cdn-webpack-plugin.
```

Comment:
```suggestion
Our build and bundling setup currently uses CommonJS modules along with webpack and the dynamic-cdn-webpack-plugin.
```

Reply:
"Utilize is not a fancy way to say use because it has a different meaning. Use means to consume from a limited supply or take something to achieve a result. Utilize means to use something beyond its intended purpose.
Use is more general and straightforward, whereas utilize implies a more strategic or efficient use of resources or opportunities"